dapp/tests/auctions/auctions.spec.ts

```diff
@@ -1,11 +1,82 @@
 // Copyright (c) 2025 IOTA Stiftung
 // SPDX-License-Identifier: Apache-2.0
 
-import { expect } from '@playwright/test';
+import { formatAddress } from '@iota/iota-sdk/utils';
 
-import { test } from '../helpers/fixtures';
+import { denormalizeName } from '@/lib/utils';
 
-test('Auction flow works', async ({ appPage }) => {
-    await expect(appPage).toHaveTitle(/IOTA Names/);
-    await expect(appPage.getByText('Your On-Chain Name')).toBeVisible();
+import { expect, test } from '../helpers/fixtures';
+import { connectWallet, createWallet, generateRandomName, requestFaucetTokens } from '../utils';
+
+test.describe.parallel('Auction Bid Flow', () => {
+    test.beforeAll(async ({ appPage, context, extensionPage, extensionName, sharedState }) => {
+        const { address, mnemonic } = await createWallet(extensionPage);
+
+        await appPage.bringToFront();
+
+        await connectWallet(appPage, context, extensionName);
+
+        await expect(appPage.getByRole('button', { name: formatAddress(address) })).toBeVisible({
+            timeout: 10_000,
+        });
+
+        await requestFaucetTokens(address);
+
+        sharedState.wallet.address = address;
+        sharedState.wallet.mnemonic = mnemonic;
+    });
+
+    test('start an aucttion', async ({ appPage: page }) => {
+        const auctionName = generateRandomName('auction');
+        const displayName = denormalizeName(auctionName);
+        console.log('displayName', displayName);
+        const initialSearch = page.getByPlaceholder('Search for your IOTA name');
+        await initialSearch.click();
+
+        const searchInput = page.getByPlaceholder('Check name availability');
+        await expect(searchInput).toBeVisible({ timeout: 15_000 });
+
+        await searchInput.fill(displayName);
+        await page.keyboard.press('Enter');
+
+        const auctionCards = page.getByTestId('auction-card');
+        await expect(auctionCards.first()).toBeVisible({ timeout: 15_000 });
+
+        const exampleCard = auctionCards.filter({
+            has: page.getByRole('heading', { name: new RegExp(`^${displayName}$`, 'i') }),
+        });
+        console.log('example', exampleCard);
+        await expect(exampleCard).toBeVisible();
+
+        await exampleCard.hover();
+        const bidButton = exampleCard.locator('button:has-text("Bid")');
+        await bidButton.waitFor({ state: 'visible', timeout: 10_000 });
+        await bidButton.click();
```

Comment:
the click function already waits for the element to be in a state visible and NOT disabled. If those checks fail, the button is not clicked and the test fails
```suggestion
        await exampleCard.getByRole('button', { name: "Bid" }).click({ timeout: 10_000 });
```